pkg/csi/service/wcpguest/controller.go

```diff
@@ -307,6 +326,121 @@ func (c *controller) ControllerPublishVolume(ctx context.Context, req *csi.Contr
 	}
 	log.Debugf("Found virtualMachine instance for node: %q", req.NodeId)
 
+	// File volumes support
+	if isFileVolumeRequest {
+		// Check the feature state for file volume support
+		if !commonco.ContainerOrchestratorUtility.IsFSSEnabled(ctx, common.FileVolume) {
+			// Feature is disabled on the cluster
+			return nil, status.Error(codes.InvalidArgument, "File volume not supported.")
+		}
+		// Build the CnsFileAccessConfig instance name and namespace
+		cnsFileAccessConfigInstance := &cnsfileaccessconfigv1alpha1.CnsFileAccessConfig{}
+		cnsFileAccessConfigInstanceName := virtualMachine.Name + "-" + req.VolumeId
+		cnsFileAccessConfigInstanceKey := types.NamespacedName{
+			Namespace: c.supervisorNamespace,
+			Name:      cnsFileAccessConfigInstanceName,
+		}
+		// Check whether the CnsFileAccessConfig instance exist in the supervisor cluster
+		if err := c.cnsOperatorClient.Get(ctx, cnsFileAccessConfigInstanceKey, cnsFileAccessConfigInstance); err != nil {
+			if !apierrors.IsNotFound(err) {
+				// Get() on the CnsFileAccessConfig instance failed with different error
+				msg := fmt.Sprintf("failed to get CnsFileAccessConfig instance: %q. Error: %+v", cnsFileAccessConfigInstance.Name, err)
+				log.Error(msg)
+				return nil, status.Errorf(codes.Internal, msg)
+			}
+			// Create the CnsFileAccessConfig instance since it is not found
+			cnsFileAccessConfigInstance = &cnsfileaccessconfigv1alpha1.CnsFileAccessConfig{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      cnsFileAccessConfigInstanceName,
+					Namespace: c.supervisorNamespace},
+				Spec: cnsfileaccessconfigv1alpha1.CnsFileAccessConfigSpec{
+					VMName:  virtualMachine.Name,
+					PvcName: req.VolumeId,
+				},
+			}
+			log.Debugf("Creating CnsFileAccessConfig instance: %+v", cnsFileAccessConfigInstance)
+			if err := c.cnsOperatorClient.Create(ctx, cnsFileAccessConfigInstance); err != nil {
+				msg := fmt.Sprintf("failed to create cnsFileAccessConfig: %v. Error: %v", cnsFileAccessConfigInstance.Name, err)
+				log.Error(msg)
+				return nil, status.Errorf(codes.Internal, msg)
+			}
+		}
+		log.Debugf("Found CnsFileAccessConfig: %+v", cnsFileAccessConfigInstance.Name)
+		if cnsFileAccessConfigInstance.DeletionTimestamp != nil {
+			// When deletionTimestamp is set, CnsOperator is in the process of removing access for this IP.
+			// When that operation is successful, the instance will be deleted. In a subsequent retry, a new instance will be created.
+			msg := fmt.Sprintf("failed to get VirtualMachines for the node: %q. Error: %+v", req.NodeId, err)
+			log.Error(msg)
+			return nil, status.Errorf(codes.Internal, msg)
+		}
+		// Verify if the CnsFileAccessConfig instance has status with done set to true and error is empty
+		if cnsFileAccessConfigInstance.Status.Done && cnsFileAccessConfigInstance.Status.Error == "" {
+			resp := &csi.ControllerPublishVolumeResponse{
+				PublishContext: cnsFileAccessConfigInstance.Status.AccessPoints,
```

Comment:
Can we set only the nfsv4 access point in the publish context?

Reply:
Makes sense